igvc_gazebo/nodes/ground_truth/main.cpp

```diff
@@ -1,32 +1,38 @@
+#include <functional>
 #include <geometry_msgs/Vector3.h>
+#include <iterator>
+#include <mutex>
 #include <nav_msgs/Odometry.h>
 #include <parameter_assertions/assertions.h>
+#include <random>
 #include <robot_localization/navsat_conversions.h>
 #include <ros/ros.h>
 #include <tf/transform_broadcaster.h>
 #include <tf/transform_datatypes.h>
 #include <tf/transform_listener.h>
-#include <mutex>
 
 ros::Publisher g_ground_truth_pub;
 // TODO make this a minimal object
 nav_msgs::Odometry g_og_pose;
 ros::Time g_last_estimate;
+double noise;
 
-void odomCallback(const nav_msgs::Odometry::ConstPtr& msg)
+void odomCallback(const nav_msgs::Odometry::ConstPtr &msg)
 {
   g_last_estimate = msg->header.stamp;
 }
 
-void groundTruthCallback(const nav_msgs::Odometry::ConstPtr& msg)
+void groundTruthCallback(const nav_msgs::Odometry::ConstPtr &msg)
 {
+  // random noise generator
+  auto dist = std::bind(std::normal_distribution<double>{ 0.0, noise }, std::mt19937(std::random_device{}()));
```

Comment:
I think this is a bit inefficient, as you're creating a new `std::random_device` every time, when you really only need it to seed the `std::mt19937` PRNG. As a whole, I think you could clean up this node by making it a class, but it's fine if you leave it as it is, it's just that with globals it's a bit more messy.
I would suggest that you put the `std::mt19937` as a global / member variable, so that you only create the `std::random_device` once, and then keep reusing the same `std::mt19937`.